src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,163 @@
+import classNames from 'classnames';
+import { useCallback } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import { SearchLink } from '../SearchLink';
+
+type Props = {
+  people: Person[]
+  selectedPerson: string
+};
+
+export const PeopleTable: React.FC<Props> = ({ people, selectedPerson }) => {
+  const activePerson = people.find(
+    person => person.slug === selectedPerson,
+  );
+
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const personsGender = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  let visiblePeople = [...people].sort((person1, person2) => {
```

Comment:
useMemo for preparing visiblePeople
and you can create custom hook which prepare this memoizated array and move logic from component

Reply:
just one question... why should i use 'useMemo' for visiblePeople if i don't pass it as a prop and it stays in the same component ?

Reply:
cause this variable will be recalculated on every rerender

Reply:
copy